### ISSUE TITLE
Include Search on documentation

### ISSUE THREAD

**Issue by sergiomcunha** [canjs/canjs#2477]
As a way to facilitate to find the pretended documentation, a search mechanism should be implemented.

**Comment by matthewp** [comment]
Search is hard to do for a statically generated site.  We might be able to leverage Google though.

**Comment by marshallswain** [comment]
I think we can wire something up with lunr.js.

**Comment by justinbmeyer** [comment]
or we could just do what the old docs have done ... we have the docMap, with everything's titles.  We can have it just search w/ that.

**Comment by justinbmeyer** [comment]
We might be able to use `lunr.js` with the docMap.  The problem is that the docMap can be quite big.

**Comment by frank-dspeed** [comment]
@justinbmeyer i would prefer if we invest in a little server for 10 dollar month and let it crawl and index the sites and implament that

but i could also think of something like a /search folder that stores the index splitted ? as we can use for example r in /search/r

**Comment by ccummings** [comment]
We could use Algolia.com which is what powers [exponentJS](https://docs.getexponent.com), [React](https://facebook.github.io/react/) as well as larger sites like HackerNews and ProductHunt.

There'd be some work on our side to make sure the docMaps we generate are indexed when we deploy the site.

We may be able to get a decent plan for free since this is for OS [src](https://www.algolia.com/doc/faq/accounts-billing/i-develop-an-open-source-or-not-for-profit-project/).

**Comment by justinbmeyer** [comment]
@mickmcgrath13 Can you update this post with where you got?

**Comment by christopherjbaker** [comment]
I wrote a quick script that [pre-builds a lunr index](https://lunrjs.com/guides/index_prebuilding.html) by crawling the generated doc folder for all html files. There are currently 1006 documents and creates an output file at 2.8mb (1.8mb if we don't index `<code>` blocks) which compresses to 459kb (or 291kb) when gzip is available. There would also need to be a map that holds info about each document (url, title, short description) (though the presence of this would be able to reduce the size of the index, as I used teh URL as the index IDs). The API is pretty straight forward if we wanted to go this route. Updates here would be part of the bit-docs command and would just deploy as before.

Algolia's free account allows 10,000 records, so we would easily fit there. For this option, we would have to re-index and upload to algolia on deploy, so making it reusable outside canjs.com would be a bit more complicated (I don't know how extensible bit-docs is).
